core/controllers/editor.py

```diff
@@ -109,8 +110,12 @@ def get(self):
         """Checks if exploration is published and redirects accordingly."""
 
         url_to_redirect_to = str(self.request.get('return_url'))
-        if url_to_redirect_to:
-            exploration_id = url_to_redirect_to.split('/')[-1]
+        url_to_redirect_to_regex = (
+            r'%s/(?P<explotation_id>[\w-]+)$' % feconf.EDITOR_URL_PREFIX)
```

Comment:
Exploration IDs are strings containing a-zA-Z0-9_-; so please make this check more strict. Also exploration is misspelled here.

Also add a backend test for a bad match.

Reply:
[\w] should take care of a-zA-Z0-9_ and i have added -. can you pls suggest other good ways.
for the rest, i will do it soon.

Reply:
Ah, sorry. I misread. What you have is fine.